src/core/proto/netlink_socket_mgr.cpp

```diff
@@ -4,173 +4,77 @@
  * SPDX-License-Identifier: GPL-2.0-only or BSD-2-Clause
  */
 
-#include "core/sock/sock-redirect.h"
 #include "core/util/utils.h"
 #include "vlogger/vlogger.h"
 #include "utils/bullseye.h"
 #include "netlink_socket_mgr.h"
 
 #include <linux/rtnetlink.h>
 #include <linux/netlink.h>
-#include <stdlib.h>
 #include <unistd.h> // getpid()
 
+#include <netlink/route/route.h>
+#include <netlink/route/rule.h>
+
 #ifndef MODULE_NAME
 #define MODULE_NAME "netlink_socket_mgr:"
 #endif
 
-#define MSG_BUFF_SIZE 81920
-
-// This function builds Netlink request to retrieve data (Rule, Route) from kernel.
-// Parameters :
-//      data_type   : either RULE_DATA_TYPE or ROUTE_DATA_TYPE
-//      pid         : opaque pid for netlink request
-//      seq         : opaque seq for netlink request
-//      buf         : buffer for the request
-//      nl_msg      : [out] pointer to request
-void netlink_socket_mgr::build_request(nl_data_t data_type, uint32_t pid, uint32_t seq, char *buf,
-                                       struct nlmsghdr **nl_msg)
-{
-    struct rtmsg *rt_msg;
-
-    assert(MSG_BUFF_SIZE >= NLMSG_SPACE(sizeof(struct rtmsg)));
-    memset(buf, 0, NLMSG_SPACE(sizeof(struct rtmsg)));
-
-    // point the header and the msg structure pointers into the buffer
-    *nl_msg = (struct nlmsghdr *)buf;
-    rt_msg = (struct rtmsg *)NLMSG_DATA(*nl_msg);
-
-    // Fill in the nlmsg header
-    (*nl_msg)->nlmsg_len = NLMSG_LENGTH(sizeof(struct rtmsg));
-    (*nl_msg)->nlmsg_seq = seq;
-    (*nl_msg)->nlmsg_pid = pid;
-    (*nl_msg)->nlmsg_type = data_type == RULE_DATA_TYPE ? RTM_GETRULE : RTM_GETROUTE;
-    (*nl_msg)->nlmsg_flags = NLM_F_DUMP | NLM_F_REQUEST;
-
-    rt_msg->rtm_family = AF_UNSPEC;
-}
-
-// Query built request and receive requested data (Rule, Route)
-// Parameters:
-//      nl_msg  : request that is built previously.
-//      buf     : [out] buffer for the reply
-//      len     : [out] length of received data.
-bool netlink_socket_mgr::query(const struct nlmsghdr *nl_msg, char *buf, int &len)
+// Update data in a table
+void netlink_socket_mgr::update_tbl(nl_data_t data_type)
 {
-    int sockfd;
-
-    // Opaque information in the request. To track expected reply.
-    uint32_t nl_pid = nl_msg->nlmsg_pid;
-    uint32_t nl_seq = nl_msg->nlmsg_seq;
+    nl_sock *sockfd = nullptr;
```

Comment:
[suggestion] lets call it just `sock`. Sockfd makes impression that this is an integer file descriptor.